### ISSUE TITLE
uWebSockets is not compilable

### ISSUE THREAD

**Issue by dnlbazac** [uWebSockets/uWebSockets#490]
Hello.

I can't install the uWebSockets library because I don't have the src/uUV.cpp file required by the CMakeList.txt file.
The compilation stops when I issue the "cmake ." command with the following output:
"-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libssl.so;/usr/lib/x86_64-linux-gnu/libcrypto.so (found version "1.0.2g") 
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.8") 
-- Configuring done
CMake Error at CMakeLists.txt:18 (add_library):
  Cannot find source file:

    src/uUV.cpp

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx

CMake Error: CMake can not determine linker language for target: uWS
CMake Error: Cannot determine link language for target "uWS".
-- Generating done
"

I am running on Ubuntu 16.04.
I am using the C++ language.
I did not make any changes to the downloaded package. It asked for the openssl library so I guess I am using SSL via uWS.
I am using the master version of uws.
The issue is reproducible all the time.

NOTE: I noticed the uUV.cpp is not on the master branch in the src folder.
I suppose it is related to the libUV library.
I removed it from the CMakeList.txt file to see if it is needed.
In this case the output of cmake is:
"
-- Configuring done
-- Generating done
-- Build files have been written to: /mnt/..."
Next the make command yielded the following error:
"CMakeFiles/echo.dir/echo.cpp.o: In function `_GLOBAL__sub_I_main':
echo.cpp:(.text.startup+0x125): undefined reference to `boost::system::generic_category()'
echo.cpp:(.text.startup+0x12a): undefined reference to `boost::system::generic_category()'
echo.cpp:(.text.startup+0x12f): undefined reference to `boost::system::system_category()'
echo.cpp:(.text.startup+0x134): undefined reference to `boost::system::system_category()'
collect2: error: ld returned 1 exit status
"
This is when I decided to ask for the experts' help.
So: wasn't the uUV.cpp file supposed to be present in the src folder?

Thank you in advance.

Daniel

**Comment by Dmitry-N-Medvedev** [comment]
Hi, Daniel.

Please, have a look at the script below. Hope that helps.

```
#!/bin/bash

cd $(mktemp -d)

ROOT_DIR=$(pwd)

LIBUV="./libuv"
UWS="./uws"

echo "INSTALLING libuv..."
gpg --keyserver pool.sks-keyservers.net --recv-keys AE9BC059
rm -rf $LIBUV > /dev/null 2>&1
git clone https://github.com/libuv/libuv.git $LIBUV
cd $LIBUV
/bin/bash autogen.sh && ./configure
make && make check && sudo make install
echo "DONE installing libuv"

echo "INSTALLING µWS..."
cd $ROOT_DIR
rm -rf $UWS > /dev/null 2>&1
git clone https://github.com/uWebSockets/uWebSockets.git $UWS
cd $UWS
cmake . && make && sudo make install
echo "DONE installing µWS"

cd ~/
rm -rf $ROOT_DIR
```

**Comment by dnlbazac** [comment]
Hello, Dimitri.

Thank you for replying so fast.
Unfortunately, this script does not help. Downloading the source code for and making libuv did not bring or create the uUV.cpp file into the src folder of the uWebSockets.
So I decided to get the v0.13.0 release. I was able to compile and install it.
I am under the impression the file uUV.cpp has been renamed to Epoll.cpp but the CMakeLists.txt still reffers top uUV.cpp.
I suppose somebody will solve this soon. on master.

Thanks again.

Daniel
